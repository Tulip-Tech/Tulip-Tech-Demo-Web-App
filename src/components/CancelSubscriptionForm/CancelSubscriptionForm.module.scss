@use 'src/styles/variables';
@use 'src/styles/theme';

.title {
  margin-bottom: 8px;
  font-weight: var(--body-font-weight-bold);
  font-size: 24px;
}

.paragraph {
  // empty
}

.confirmButton {
  margin-bottom: 8px;
}
