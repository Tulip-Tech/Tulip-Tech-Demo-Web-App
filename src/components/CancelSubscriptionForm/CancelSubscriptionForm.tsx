import React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './CancelSubscriptionForm.module.scss';

import Button from '#components/Button/Button';
import FormFeedback from '#components/FormFeedback/FormFeedback';

type Props = {
  onConfirm: () => void;
  onCancel: () => void;
  error: string | null;
  submitting: boolean;
};

const CancelSubscriptionForm: React.FC<Props> = ({ onConfirm, onCancel, error, submitting }: Props) => {
  const { t } = useTranslation('account');

  return (
    <div>
      {error ? <FormFeedback variant="error">{error}</FormFeedback> : null}
      <h2 className={styles.title}>{t('cancel_subscription.title')}</h2>
      <p className={styles.paragraph}>{t('cancel_subscription.explanation')}</p>
      <Button
        className={styles.confirmButton}
        label={t('cancel_subscription.unsubscribe')}
        color="primary"
        variant="contained"
        onClick={onConfirm}
        fullWidth
        disabled={submitting}
      />
      <Button label={t('cancel_subscription.no_thanks')} variant="outlined" onClick={onCancel} fullWidth />
    </div>
  );
};

export default CancelSubscriptionForm;
