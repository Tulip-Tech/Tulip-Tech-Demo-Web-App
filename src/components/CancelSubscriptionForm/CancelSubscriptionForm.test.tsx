import React from 'react';
import { render } from '@testing-library/react';

import CancelSubscriptionForm from './CancelSubscriptionForm';

describe('<CancelSubscriptionForm>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<CancelSubscriptionForm submitting={false} error={null} onCancel={vi.fn()} onConfirm={vi.fn()} />);

    expect(container).toMatchSnapshot();
  });
});
