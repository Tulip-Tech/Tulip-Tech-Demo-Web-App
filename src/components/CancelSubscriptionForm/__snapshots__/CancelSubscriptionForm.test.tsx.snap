// Vitest Snapshot v1

exports[`<CancelSubscriptionForm> > renders and matches snapshot 1`] = `
<div>
  <div>
    <h2
      class="_title_dba43b"
    >
      cancel_subscription.title
    </h2>
    <p>
      cancel_subscription.explanation
    </p>
    <button
      aria-disabled="false"
      class="_button_f8f296 _confirmButton_dba43b _primary_f8f296 _fullWidth_f8f296"
    >
      <span>
        cancel_subscription.unsubscribe
      </span>
    </button>
    <button
      class="_button_f8f296 _default_f8f296 _outlined_f8f296 _fullWidth_f8f296"
    >
      <span>
        cancel_subscription.no_thanks
      </span>
    </button>
  </div>
</div>
`;
